src/cgr_gwas_qc/workflow/sub_workflows/contamination.smk

```diff
@@ -67,195 +65,249 @@ rule verifyidintensity_conda:
-        resources:
-            mem_gb=lambda wildcards, attempt: attempt * 4,
-            time_hr=lambda wildcards, attempt: attempt ** 2,
-        script:
-            "../scripts/agg_verifyidintensity.py"
-
+# If BCF file is input
+if cfg.config.user_files.bcf:
+    if config.get("cluster_mode", False):
+
+        localrules:
+            agg_verifyidintensity,
+
+        rule grouped_contamination:
+            """Extracts normalized intensities and other metrics from an aggregated
+            VCF file and writes to Illumina ADPC.BIN per sample.
+
+            This is the format required by ``verifyIDintensity``. The script also
+            runs some sanity checks (intensities and normalized intensities > 0;
+            genotypes are one of {0, 1, 2, 3}) while processing each file.
+
+            .. warning::
+                This is a submission hot spot creating 1 job per sample.
+
+            Find contaminated samples using allele intensities.
+
+            Uses ``verifyIDintensity`` to find samples with allele intensities that deviate from the
+            population.
+
+            .. warning::
+                This is a submission hot spot creating 1 job per sample.
+
+            .. note::
+                Here we are running ``verifyIDintensity`` in single sample mode. This software also has a
+                multi-sample mode which may be faster and give better estimates. The problem with
+                multi-sample mode is that it only works when you have a "large" number of samples.
+            """
+            input:
+                sample_sheet_csv="cgr_sample_sheet.csv",
+                vcf_file=cfg.config.user_files.bcf,
+                abf_file=rules.pull_b_allele_freq_from_1kg.output.abf_file,
+                _=rules.verifyidintensity_conda.output[0],
+            params:
+                grp="{grp}",
+                snps=cfg.config.num_snps,
+                conda_env=cfg.conda("verifyidintensity"),
+                notemp=config.get("notemp", False),
+            output:
+                temp("sample_level/contamination/{grp}/verifyIDintensity.csv"),
+            threads: 12
+            resources:
+                mem_mb=lambda wildcards, attempt: 1024 * 12 * attempt,
+                time_hr=lambda wildcards, attempt: 4 * attempt,
+            script:
+                "../scripts/grouped_contamination.py"
+
+        rule agg_verifyidintensity:
+            input:
+                expand(rules.grouped_contamination.output[0], grp=cfg.cluster_groups),
+            output:
+                "sample_level/contamination/verifyIDintensity.csv",
+            resources:
+                mem_gb=lambda wildcards, attempt: attempt * 4,
+                time_hr=lambda wildcards, attempt: attempt**2,
+            script:
+                "../scripts/agg_verifyidintensity.py"
+
+    else:
+
+        rule per_sample_vcf_to_adpc:
+            """Converts a sample's GTC/BPM to an Illumina ADPC.BIN.
```

Comment:
This docstring needs to be updated to reflect that the input is a BCF not GTC/BPM.

Reply:
Updated in f6ed2d7bf1f5213b6a1f5b793c3cd0c4321f0e5d